core/src/main/java/io/keyko/monitoring/agent/core/chain/block/ViewBlockListener.java

```diff
@@ -0,0 +1,129 @@
+import io.keyko.monitoring.agent.core.service.exception.NotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.web3j.abi.datatypes.Function;
+import org.web3j.abi.datatypes.Type;
+
+import java.math.BigInteger;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A block listener that manage the view requests after blocks are mined
+ *
+ */
+@Component
+public class ViewBlockListener implements BlockListener {
+
+    private static final Logger log = LoggerFactory.getLogger(ViewBlockListener.class);
+
+    private BlockchainService blockchainService;
+    private BlockchainEventBroadcaster eventBroadcaster;
+    private Map<String, ContractViewFilter> viewFilters= new HashMap<>();
+    private NodeSettings nodeSettings;
+    private Web3jEventParameterConverter eventParameterConverter;
+    private ContractViewDetailsFactory contractViewDetailsFactory;
+
+    public ViewBlockListener(BlockchainService blockchainService,
+                             BlockchainEventBroadcaster eventBroadcaster,
+                             NodeSettings nodeSettings,
+                             Web3jEventParameterConverter eventParameterConverter,
+                             ContractViewDetailsFactory contractViewDetailsFactory) {
+
+        this.blockchainService = blockchainService;
+        this.eventBroadcaster = eventBroadcaster;
+        this.nodeSettings = nodeSettings;
+        this.eventParameterConverter= eventParameterConverter;
+        this.contractViewDetailsFactory = contractViewDetailsFactory;
+    }
+
+    @Override
+    public void onBlock(Block block) {
+        log.info(String.format("New block mined. Hash: %s, Number: %s, Filters: %s",
+                block.getHash(), block.getNumber(), viewFilters.size()));
+
+        // Foreach registered filter
+        viewFilters.forEach((id, filter) -> {
+            BigInteger interval= BigInteger.valueOf(
+                    filter.getPollingStrategy().getBlockInterval().intValue());
+
+            // Check if blockInterval requires req
+            BigInteger mod= block.getNumber().mod(interval);
+
+            if (mod.equals(BigInteger.ZERO))    {
+                processViewMessage(filter, block);
+            }
+        });
+    }
+
+    private Function composeFunction(ContractViewFilter filter) {
+        return filter.getMethodSpecification().getWeb3Function();
+    }
+
+    public void addViewFilter(ContractViewFilter filter) {
+        log.debug(String.format(
+                "Adding new filter to ViewBlockListener: %s", filter.getId()));
+        if (!viewFilters.containsKey(filter.getId()))
+            viewFilters.put(filter.getId(), filter);
+    }
+
+    public boolean processViewMessage(ContractViewFilter filter, Block block)  {
+        // Compose message
+        Function _func= filter.getMethodSpecification().getWeb3Function();
```

Comment:
Do make sense use the `composeFunction` method declare few lines before?

Reply:
:+1: changed